packages/cli/src/commands/init.ts

```diff
@@ -148,40 +161,83 @@ async function loadDependencies(): Promise<Record<string, string>> {
     //   packageJson.dependencies[dependency]
     // );
-    packageJson.dependencies[dependency] = normalizeVersion(packageJson.dependencies[dependency]);
+    packageJson.dependencies[dependency] = normalizeVersion(
+      packageJson.dependencies[dependency]
+    );
   }
   return packageJson.dependencies;
 }
 
-async function config() {
-    const { confirmation } = await prompt({
-      type: "toggle",
-      name: "confirmation",
-      message: prompts.writeConfiguration,
-      initial:true,
-      active: "yes",
-      inactive: "no",
-    })
-    if (confirmation) {
-      if (!fs.existsSync(COMPONENTS_JSON_PATH)) {
-        // technically this may not be necessary. as the isInitialized fn will check if the file exists.
-        ora(prompts.shadcnRequired).fail();
-        process.exit(1);
-      }
-      const componentsJson = parseComponentsJson();
+const TSCONFIG_JSON_PATH = path.join(process.cwd(), "tsconfig.json");
+export function parseTsconfigJson() {
+  if (fs.existsSync(TSCONFIG_JSON_PATH)) {
+    return JSON.parse(fs.readFileSync(TSCONFIG_JSON_PATH, "utf-8"));
+  } else {
+    return {};
+  }
+}
 
-      // Create trnsprncy alias
-      componentsJson.aliases.trnsprncy = TRNSPRNCY_PATH;
+function hasSrcPath(): boolean {
+  try {
+    const tsconfig = parseTsconfigJson();
+    const paths = tsconfig.compilerOptions?.paths || {};
+    return !!paths["@/*"] && paths["@/*"][0] === "./src/*";
+  } catch (error) {
+    console.error("Error parsing tsconfig:", error);
+    return false;
+  }
+}
 
-      fs.writeFileSync(
-        COMPONENTS_JSON_PATH,
-        JSON.stringify(componentsJson, null, 2)
-      );
-      ora(prompts.configurationWritten).succeed();
-    } else {
-      ora(prompts.operationAborted).fail();
+const mkdir_components = (path: string) => {
+  fs.mkdir(path, { recursive: true }, (err) => {
+    if (err) {
+      console.error("Error creating directory:", err);
     }
-  
+  });
+};
+
+const decide = {
+  true: () =>
+    mkdir_components(
+      path.join(process.cwd(), "/src", TRNSPRNCY_PATH.replace("@", ""))
+    ),
+  false: () =>
+    mkdir_components(path.join(process.cwd(), TRNSPRNCY_PATH.replace("@", ""))),
+};
+
+async function config() {
+  const { confirmation } = await prompt({
+    type: "toggle",
+    name: "confirmation",
+    message: prompts.writeConfiguration,
+    initial: true,
+    active: "yes",
+    inactive: "no",
+  });
+  if (confirmation) {
+    if (!fs.existsSync(COMPONENTS_JSON_PATH)) {
+      // technically this may not be necessary. as the isInitialized fn will check if the file exists.
+      ora(prompts.shadcnRequired).fail();
+      process.exit(1);
+    }
+    const componentsJson = parseComponentsJson();
+
+    // Create trnsprncy alias
+    componentsJson.aliases.trnsprncy = TRNSPRNCY_PATH;
+
+    fs.writeFileSync(
+      COMPONENTS_JSON_PATH,
+      JSON.stringify(componentsJson, null, 2)
+    );
+
+    const srcPath = hasSrcPath() ? "true" : "false";
```

Comment:
Why are we assigning strings? shouldn't we just assing the boolean values? 

```suggestion
    const srcPath = hasSrcPath() ? true : false;
```

Reply:
It seems we cannot use Boolean values as kays for a dictionary

Reply:
After discussing, it seems this was a conscious decision, and the correct decision in this case.